docs/proposals/mcs/remove-derived-mcs.md

```diff
@@ -0,0 +1,352 @@
+  ....
+    Corefile: |
+     .:53 {
+         errors
+         health {
+            lameduck 5s
+         }
+         ready
+         multicluster clusterset.local # Add this line.
+         kubernetes cluster.local in-addr.arpa ip6.arpa {
+            pods insecure
+            fallthrough in-addr.arpa ip6.arpa
+            ttl 30
+         }
+         prometheus:9153
+         forward ./etc/resolv.conf {
+            max_concurrent 1000
+         }
+         cache 30
+         the loop
+         reload
+         load balance
+     }
+...
+```
+
+* Configure dns search item "clusterset.local"
+
+**Configure node /etc/resolv.conf file**
+
+Edit the `/etc/resolv.conf` file and add the "clusterset.local" entry to give priority to local services when using the domain name `foo.default.svc`.
+
+```shell
+root@btg:/# cat /etc/resolv.conf
+search default.svc.cluster.local svc.cluster.local cluster.local clusterset.local
+nameserver 10.13.0.10
+options ndots: 5
+```
+
+This configuration method needs to configure the `/etc/resolv.conf` file on the client node. If you don't want to change the node's `resolv.conf` file, you can also use the following method.
+
+**Configure pod's dnsConfig in yaml**
+
+That is, add the configuration item searches of dnsConfig in yaml, and set the value to `clusterset.local`, as shown below:
+
+```yaml
+...
+       dnsPolicy: ClusterFirst
+       dnsConfig:
+         searches:
+           -clusterset.local
+         options:
+           - name: timeout
+             value: "2"
+```
+
+###### Access method
+
+Since the search function of dns is used, the `foo` service  access method is:
+
+1. `foo.default.svc` access. This method gives priority to local access, and if the local service with the same name is unavailable, it will access the remote cluster service.
+
+2. `foo.default.svc.clusterset.local` full domain name access. The request will go directly to the remote server for processing.
+
+###### Advantages and disadvantages
+
+shortcoming:
+
+1. Using `foo.default.svc` access cannot achieve load balancing. The remote service will only be accessed when the local service is unavailable. Therefore, if you need to achieve local and remote access load balancing, you need to cooperate with other solutions.
+
+2. To use `foo.default.svc`, you need to modify the dns configuration. If you use the `resolv.conf `method, you need to modify the `resolv.conf` file of each possible nodes. If you use the dnsConfig method, you need to modify the yaml or api of the delivered pod.
+
+3. The `multicluster` plug-in cannot be loaded dynamically. Customers need to recompile `coredns` and modify the `coredns` configuration of member clusters.
+
+4. The original access method will be changed. This solution uses different suffixes to distinguish local and remote services. Therefore, the original access method `foo.default.svc.cluster.local` will become `foo.default.svc.clusterset.local`. Customers need to modify the code to adapt.
```

Comment:
The distinction should be between single-cluster Service and multi-cluster Service.

Reply:
ok